app/models/mixins/per_ems_worker_mixin.rb

```diff
@@ -97,7 +97,7 @@ def queue_name_for_ems(ems)
       return "generic" if ems.kind_of?(Host) && ems.acts_as_ems?
 
       return ems unless ems.kind_of?(ExtManagementSystem)
```

Comment:
@jrafanie this looks like the only caller that implies the argument wouldn't be an ems, https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/per_ems_worker_mixin.rb#L66-L67

Maybe `return ems if ems.kind_of?(String)` would be better if it is even still needed?